enigma-principal/app/src/epoch_u/epoch_provider.rs

```diff
@@ -216,19 +261,44 @@ impl EpochProvider {
 //////////////////////// TESTS  /////////////////////////////////////////
 
 #[cfg(test)]
-mod test {
+pub mod test {
+    use std::collections::HashMap;
+
     use web3::types::{Bytes, H160};
 
     use enigma_crypto::KeyPair;
+    use enigma_tools_u::{esgx::general::storage_dir};
     use enigma_types::ContractAddress;
 
     use super::*;
 
     pub const WORKER_SIGN_ADDRESS: [u8; 20] =
         [95, 53, 26, 193, 96, 206, 55, 206, 15, 120, 191, 101, 13, 44, 28, 237, 80, 151, 54, 182];
 
+    pub fn setup_epoch_storage() {
```

Comment:
What is this complicated thing actually do? why no just create the dir?
and this runs *only* in tests. which means that if the directory isn't existing yet in real code then this won't work.
what do you think about this fix? https://github.com/enigmampc/enigma-core/pull/132

Reply:
I replied to #132 about why this is only an issue in tests. This "complicated thing" creates the directory and ensures that it is empty (so a test does not accidentally read the state from another).